lib/MySQL_Monitor.cpp

```diff
@@ -480,35 +483,170 @@ void MySQL_Monitor_Connection_Pool::put_connection(char *hostname, int port, MYS
+			mysql_thread___monitor_replication_lag_use_percona_heartbeat[0] != '\0') {
+			use_percona_heartbeat = true;
+			query_ = "SELECT MAX(ROUND(TIMESTAMPDIFF(MICROSECOND, ts, SYSDATE(6))/1000000)) AS Seconds_Behind_Master FROM ";
+			query_ += mysql_thread___monitor_replication_lag_use_percona_heartbeat;
+		} else {
+			query_ = "SHOW SLAVE STATUS";
+		}
+		task_timeout_ = mysql_thread___monitor_replication_lag_timeout;
+		task_handler_ = &MySQL_Monitor_State_Data::replication_lag_handler;
+		break;
+	case MON_GALERA:
+		async_state_machine_ = ASYNC_QUERY_START;
+#ifdef TEST_GALERA
+		query_ = "SELECT wsrep_local_state , read_only , wsrep_local_recv_queue , wsrep_desync , wsrep_reject_queries , wsrep_sst_donor_rejects_queries , "
+			" wsrep_cluster_status, pxc_maint_mode FROM HOST_STATUS_GALERA WHERE hostgroup_id=";
+		query_ += std::to_string(writer_hostgroup) + " AND hostname='" + std::string(hostname) + "' AND port=" + std::to_string(port);
+#else
+		if (strncmp(mysql->server_version, (char*)"5.7", 3) == 0 || strncmp(mysql->server_version, (char*)"8", 1) == 0) {
+			// the backend is either MySQL 5.7 or MySQL 8 : INFORMATION_SCHEMA.GLOBAL_STATUS is deprecated
+			query_ = "SELECT (SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME='WSREP_LOCAL_STATE') "
+				"wsrep_local_state, @@read_only read_only, (SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME='WSREP_LOCAL_RECV_QUEUE') wsrep_local_recv_queue , "
+				"@@wsrep_desync wsrep_desync, @@wsrep_reject_queries wsrep_reject_queries, @@wsrep_sst_donor_rejects_queries wsrep_sst_donor_rejects_queries, "
+				"(SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME='WSREP_CLUSTER_STATUS') wsrep_cluster_status , "
+				"(SELECT COALESCE(MAX(VARIABLE_VALUE),'DISABLED') FROM performance_schema.global_variables WHERE variable_name='pxc_maint_mode') pxc_maint_mode ";
+		} else {
+			// any other version
+			query_ = "SELECT (SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='WSREP_LOCAL_STATE') "
+				"wsrep_local_state, @@read_only read_only, (SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='WSREP_LOCAL_RECV_QUEUE') wsrep_local_recv_queue , "
+				"@@wsrep_desync wsrep_desync, @@wsrep_reject_queries wsrep_reject_queries, @@wsrep_sst_donor_rejects_queries wsrep_sst_donor_rejects_queries, "
+				"(SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='WSREP_CLUSTER_STATUS') wsrep_cluster_status , (SELECT 'DISABLED') pxc_maint_mode";
+		}
+#endif // TEST_GALERA
+		task_timeout_ = mysql_thread___monitor_galera_healthcheck_timeout;
+		task_handler_ = &MySQL_Monitor_State_Data::galera_handler;
+		break;
+	case MON_CLOSE_CONNECTION:
+		break;
+	case MON_CONNECT:
+		break;
+	case MON_AWS_AURORA:
+		break;
+	}
+}
+
+void MySQL_Monitor_State_Data::mark_task_as_timeout(unsigned long long time) {
+	
```

Comment:
There are several places that hold this extra empty lines after declarations or if statements, but only happens sometimes, I don't know if there is a reason for them. Just mentioning it in case it's an editor thing, because they look "random".

Reply:
It's done intentionally to avoid code clutter.